Proposal to update the TSC terms (#4686)
kedro-org/kedro: docs/about/technical_steering_committee.md

# Kedro's Technical Steering Committee

> Kedro is a graduated project within [LF AI & Data](https://lfaidata.foundation/).

Reviewer: ```suggestion
Kedro is a graduate-stage project within [LF AI & Data](https://lfaidata.foundation/).
```
Nit: I think this is the right terminology; see https://lfaidata.foundation/projects/kedro/.

### Advisors

Advisors are individuals who are not actively contributing to the codebase or attending regular meetings but provide valuable expertise, feedback, and strategic guidance to the Kedro project.

## Responsibilities of a voting member

- Defining and maintaining the product vision and roadmap
- Supporting key technical and strategic initiatives
- Providing input on complex or high-impact technical decisions
- Advising and mentoring contributors
- Representing Kedro within the broader community
- Attend regular team meetings to discuss the project plans and roadmap

### Voting rights
Voting members take part in two types of votes:

- Code modifications – follows the [KEP process](https://github.com/kedro-org/kedro/discussions/5150)
- Membership additions – new voting members must receive majority approval (≥50%) from existing voting members (see more details below)

## Process to become a voting member

> Just contributing does not make you a voting member. You need to show commitment to Kedro's long-term success by working with existing TSC members for at least one year.

Reviewer: I think they should be promoted from one of the existing non-voting roles? As in, it doesn't make sense to me to become a voting member directly, but rather to have been part of the TSC for at least one year prior (not just working with them).

Reviewer 2: I agree with @deepyaman , internal org structures for McKinsey shouldn't dictate how an open project operate. Probably 12 months is too long though, as I imagine that six months should be enough for someone to be committed for the long-term success of the project.

Reviewer 3: Thanks all! After another quick discussion we feel this makes sense. I'll update the text accordingly.

Any current voting member may nominate a new candidate and will serve as their advocate, responsible for initiating and managing the voting process. The nominating member should open a pull request and formally call for a vote on the candidate, posting the announcement in the #kedro-tsc channel on the Kedro Slack organisation.

The voting period should remain open for at least two weeks, and the proposal will pass with a majority approval. Once voting concludes, the #kedro-tsc channel should be updated with the final results.

The new maintainer should then be added to the `kedro-developers` team on the Kedro GitHub organisation,
the `kedro-tsc` channel on the Kedro Slack organisation, the regular TSC meetings, and the `CITATION.cff` file.

### Inactivity and removal
Maintainers who do not contribute code, documentation, or attend regular meetings for 6 months will be automatically removed from the [current TSC members](#current-kedro-tsc-members) list, `kedro-developers` team, and `CITATION.cff` file. They may be invited to join the [advisor](#advisors) role if they are interested in continuing to support and contribute to the project in a different capacity.

> ## Responsibilities of an advisor

Reviewer 3: Yes it gives them access to `kedro-tsc` channel and invites to meetings. I also think this lowers the barrier or entry for people who'd like to be involved but can't guarantee any time commitment.

> - Providing ongoing, high-level guidance on direction, governance, and technical topics (rather than ad-hoc support)

Reviewer: To play the devil's advocate: as a maintainer who has been contributing to the project for the past many years, why would I look for ongoing guidance on direction, governance, etc. from somebody who has not put the same time and effort into Kedro?  I would expect this guidance to come from the voting members of the TSC (and, conversely, voting members should be able to provide this sort of direction).

Reviewer 4: By advisors, we mean people who are very knowledgeable in this space and work in MLOps or GenAI on a regular basis. These are practitioners whose day-to-day job is to put data pipelines into production and operate them in real environments.

The Kedro team (at McK) is largely made up of software engineers, and most of us don’t use Kedro or similar tools in production every day. Advisors bring that practitioner perspective and a different lens from those building the product day to day similar to how advisory boards work in startups.

- Acting as external ambassadors or thought partners
- Sharing lessons learned and best practices from adjacent domains

## Process to become an advisor
Any current voting member may nominate a new candidate and will serve as their advocate, responsible for initiating and managing the voting process. The nominating member should open a pull request and formally call for a vote on the candidate, posting the announcement in the #kedro-tsc channel on the Kedro Slack organisation.

The voting period should remain open for at least two weeks, and the proposal will pass with a majority approval. Once voting concludes, the #kedro-tsc channel should be updated with the final results.

### Inactivity and removal

> Advisors may be automatically removed after 12 months of inactivity. Inactivity is defined as a lack of any engagement with Kedro during this period.

Reviewer 3: We discussed this as well and ended up setting it to 12 months because the expectation is that advisors might not have the same capacity as more active TSC member roles and therefore it's more likely they would be MIA for longer periods at a time.

> Advisors may be automatically removed after 12 months of inactivity. Inactivity is defined as a lack of any engagement with Kedro during this period.

Reviewer 2: I think 6 months should be long enough, if someone is MIA for even minimal activity for more than 6 months, it's fair they aren't considered advisors. They can always come back to the role once they free up more time and request to be included here. Btw this raises the question for me, how are those automated expiries going to be enforced? I don't imagine anyone will count activity for each individual member of the TSC.

### Maintainers

Maintainers have repository access and full participation rights in discussions but do not hold voting privileges. They are active contributors who focus on different areas of the project. Maintainers are responsible for maintaining the health of the project, participating in planning and (technical) design decisions, and otherwise acting as part of the core Kedro development team.

### Advisors

Advisors are individuals who are not actively contributing to the codebase or attending regular meetings but provide valuable expertise, feedback, and strategic guidance to the Kedro project.

## Responsibilities of a voting member

- Defining and maintaining the product vision and roadmap
- Supporting key technical and strategic initiatives
- Providing input on complex or high-impact technical decisions
- Advising and mentoring contributors
- Representing Kedro within the broader community
- Attend regular team meetings to discuss the project plans and roadmap

### Voting rights
Voting members take part in two types of votes:

- Code modifications – follows the [KEP process](https://github.com/kedro-org/kedro/discussions/5150)

> - Membership additions – new voting members must receive majority approval (≥50%) from existing voting members (see more details below)

Reviewer 3: We discussed this and basically the justification for that is:

1. that is the current process described in the [Kedro Technical Charter](https://github.com/kedro-org/kedro/blob/main/kedro_technical_charter.pdf) section 3c (screenshot below)
   a. NB: it is only for votes on changing the charter that 2/3 majority are required, in section 8a (screenshot below)
2. it aligns with the [LF AI & Data Charter](https://lfaidata.foundation/wp-content/uploads/sites/3/2025/09/LF-AI-Data-Foundation-Participation-Agreement-Charter-Sep.-2025.pdf) section 8 c (which mirrors the exact wording currently used in the Kedro Technical Charter):
   a. _"Except as provided in Section 16.a., decisions by vote at a meeting will require a simple majority vote, provided quorum is met. Except as provided in Section 16.a., decisions by electronic vote without a meeting will require a majority of all voting representatives."_
3. it seems to align with the "procedural" voting [processes of the Apache foundation](https://www.apache.org/foundation/voting.html) as well (directly complementary to the KEP model we adopted), where it says "Votes on procedural issues follow simple majority unless otherwise stated."

<img width="1420" height="200" alt="image (16)" src="https://github.com/user-attachments/assets/935e1196-1afb-43a9-bc07-a078aebbcaf4" />
<img width="1562" height="216" alt="image (17)" src="https://github.com/user-attachments/assets/481fe8b9-12b9-460b-8d37-4cb4157d1fb7" />

cc: @Joseph-Perkins @AliceCima10 @rashidakanchwala

## Requirements to become a maintainer

Just contributing does not make you a maintainer; you need to demonstrate commitment to Kedro's long-term success by
working with existing maintainers for at least one year.

On top of the [committer responsibilities](#responsibilities-of-a-committer), we look for people who can do at least some of the following:

- Steer the direction of the project to ensure it retains relevance, including by bringing awareness of industry trends to the team and connecting the project to complementary initiatives
- Give talks about the project at conferences and online events, write blog posts, or other forms of advocacy
- Show excitement about the future of Kedro

> - In case of contributors from the community who haven't been previously appointed as committers,
>   show evidence of already having started pull requests and code reviews under the guidance of maintainers;
>   including asking for help where needed

Reviewer: ```suggestion
```
Is there any real reason for this carve-out? Contributor -> committer -> maintainer seems standard enough.

Appointing new maintainers requires a vote of the TSC,
described in [Adding or removing maintainers](#adding-or-removing-maintainers).

## Responsibilities of a committer

Committers are observer (non-voting) members of the TSC. They are active contributors who are responsible for maintaining the health of the project, participating in planning and technical design decisions, and otherwise acting as part of the core Kedro development team.

We look for people who can do at least some of the following:

- Write high-quality code and collaborate with the team and community
- Make substantial improvements to our documentation, either in terms of content or information architecture
- Improve the aesthetics and usability of the project, including its documentation, website, and user interfaces (graphical and textual)
- Understand the project's code base and internals
- Develop valuable plugins that complement the functionality of Kedro

> - Make pull requests from our backlog or roadmap; maintainers need to work towards a common goal

Reviewer 3: ?
```suggestion
- Make pull requests from our backlog or roadmap; committers need to work towards a common goal
```

[Vladimir Nikolic](https://github.com/vladimir-mck) and
[Zain Patel](https://github.com/mzjp2).

## Application process for maintainers

> At any point, existing maintainers can propose that a contributor who fulfills the [requirements to become a maintainer](#requirements-to-become-a-maintainer) is promoted to maintainer. A TSC [vote](#voting-process) is required.

Reviewer: ```suggestion
At any point, existing maintainers can propose that a committer who fulfills the [requirements to become a maintainer](#requirements-to-become-a-maintainer) is promoted to maintainer. A TSC [vote](#voting-process) is required.
```
In line with the above suggestion
